### ISSUE TITLE
Unhandled Rejection (SecurityError): The operation is insecure.

### ISSUE THREAD

**Issue by Hardik500** [badges/shields#3901]
Are you experiencing an issue with...

- [ ] [shields.io](https://shields.io/#/)
- [X] My own instance
- [ ] [gh-badges NPM package](https://www.npmjs.com/package/gh-badges)

:beetle: **Description**

I just started running the development using the steps in TUTORIAL.md after running `npm start`. After that the browser tries to render the webpage but provides an error,
Here is the screen shot file:

![Screenshot_1](https://user-images.githubusercontent.com/28431356/63434359-8df46300-c442-11e9-8ecc-b964179523e8.png)

The logs produced were:
`
[server] (node:10068) UnhandledPromiseRejectionWarning: Error: Token pool is exhausted
[server]     at TokenPool._nextBatch (C:\Users\User\Documents\User\git\shields\core\token-pooling\token-pool.js:263:11)
[server]     at TokenPool.next (C:\Users\User\Documents\User\git\shields\core\token-pooling\token-pool.js:300:20)
[server]     at GithubApiProvider.tokenForUrl (C:\Users\User\Documents\User\git\shields\services\github\github-api-provider.js:155:33)
[server]     at GithubApiProvider.request (C:\Users\User\Documents\User\git\shields\services\github\github-api-provider.js:171:22)
[server]     at Promise (C:\Users\User\Documents\User\git\shields\services\github\github-api-provider.js:210:12)
[server]     at new Promise (<anonymous>)
[server]     at GithubApiProvider.requestAsPromise (C:\Users\User\Documents\User\git\shields\services\github\github-api-provider.js:209:12)
[server]     at GithubForks._requestFetcher (C:\Users\User\Documents\User\git\shields\services\github\github-auth-service.js:13:23)
[server]     at GithubForks._request (C:\Users\User\Documents\User\git\shields\core\base-service\base.js:223:40)
[server]     at GithubForks._requestGraphql (C:\Users\User\Documents\User\git\shields\core\base-service\base-graphql.js:72:35)
[server] (node:10068) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
[server] (node:10068) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
`

:link: **Link to the badge**

:bulb: **Possible Solution**

**Comment by paulmelnikow** [comment]
Hi! Could this be the issue? https://stackoverflow.com/a/49547221/893113 Maybe try it in Chrome?

**Comment by Hardik500** [comment]
It seems the problem exists on firefox only since I tried it on brave and it worked perfectly but I noticed two things:
1.  The log is still the same no matter what browser I open
2.  Even if I disabled cookies( which I did on brave ) it still works so the problem is not with the cookies. And I even unblocked all cookies on firefox but it still gives me the same error.

**Comment by paulmelnikow** [comment]
Ah, the logs are unrelated to the security error.

**Comment by Hardik500** [comment]
But the problem of it not running with firefox still exists.

**Comment by paulmelnikow** [comment]
I can't reproduce it in my copy of Firefox (68.0.1). Are you installing dependencies with `npm ci`?That would ensure you have the same versions as we have.

The stack trace you're seeing isn't from Shields, so this is either an underlying issue in Gatsby, or something in your Firefox setup.

If you run the built site, does the problem go away? Try `npm run build` and then `npm run start:server` and then open `http://localhost:8080`.

**Comment by Hardik500** [comment]
I tried the steps which you described, but after running `npm run build` I got a new error in my console

```
internal/modules/cjs/loader.js:583
    throw err;
    ^

Error: Cannot find module 'is-css-color'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (C:\Users\User\Documents\User\git\shields\gh-badges\lib\color.js:3:20)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! shields.io@0.0.0 defs: `node scripts/export-service-definitions-cli.js > service-definitions.yml`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the shields.io@0.0.0 defs script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.       

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\User\AppData\Roaming\npm-cache\_logs\2019-08-24T10_33_42_357Z-debug.log
ERROR: "defs" exited with 1.
```

**Comment by paulmelnikow** [comment]
Ah, yea, that's another issue we're trying to solve at the moment; see #3904.

**Comment by Hardik500** [comment]
So is there any workaround or I need to wait for an update.

**Comment by paulmelnikow** [comment]
I think `npm install` in the `gh-badges` folder?

**Comment by Hardik500** [comment]
It works for now but still the same error on firefox still persists.
Anyway should I close the issue for now?

**Comment by paulmelnikow** [comment]
Sure, sounds good. I don’t have the capacity to investigate the Firefox issue, especially since there’s a workaround.

**Comment by paulmelnikow** [comment]
It looks like the workaround for the `is-css-color` issue is to `npm install -g npm@6.10.3`. There's what I think is a regression in 6.11, probably this issue: https://npm.community/t/6-11-1-some-dependencies-are-no-longer-being-installed/9586.
